src/blocks/uptime.rs

```diff
@@ -0,0 +1,116 @@
+use std::time::Duration;
+use chan::Sender;
+use block::{Block, ConfigBlock};
+use config::Config;
+use de::deserialize_duration;
+use errors::*;
+use widgets::text::TextWidget;
+use widget::I3BarWidget;
+use scheduler::Task;
+use uuid::Uuid;
+use blocks::lib::*;
+
+pub struct Uptime {
+    text: TextWidget,
+    id: String,
+    update_interval: Duration,
+
+    //useful, but optional
+    #[allow(dead_code)] config: Config,
+    #[allow(dead_code)] tx_update_request: Sender<Task>,
+}
+
+#[derive(Deserialize, Debug, Default, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct UptimeConfig {
+    /// Update interval in seconds
+    #[serde(default = "UptimeConfig::default_interval", deserialize_with = "deserialize_duration")]
+    pub interval: Duration,
+}
+
+impl UptimeConfig {
+    fn default_interval() -> Duration {
+        Duration::from_secs(60)
+    }
+}
+
+impl ConfigBlock for Uptime {
+    type Config = UptimeConfig;
+
+    fn new(block_config: Self::Config, config: Config, tx_update_request: Sender<Task>) -> Result<Self> {
+        Ok(Uptime {
+            id: Uuid::new_v4().simple().to_string(),
+            update_interval: block_config.interval,
+            text: TextWidget::new(config.clone()),
+            tx_update_request: tx_update_request,
+            config: config,
+        })
+    }
+}
+
+impl Block for Uptime {
+    fn update(&mut self) -> Result<Option<Duration>> {
+        let uptime_raw = match read_file("uptime", "/proc/uptime") {
+            Ok(file) => file,
+            Err(e) => {
+                return Err(BlockError(
+                    "Uptime".to_owned(),
+                    format!("Uptime failed to read /proc/uptime: '{}'", e),
+                ));
+            }
+        };
+        let uptime = match uptime_raw.split_whitespace().nth(0) {
+            Some(uptime) => uptime,
+            None => {
+                return Err(BlockError(
+                    "Uptime".to_owned(),
+                    "Uptime failed to read uptime string.".to_owned(),
+                ));
+            }
+        };
+
+        let total_seconds = match uptime.parse::<f64>() {
+            Ok(uptime) => uptime as u32,
+            Err(e) => {
+                return Err(BlockError(
+                    "Uptime".to_owned(),
```

Comment:
For consistency, can you downcase this and the error above to match other blocks? And there's no need to prefix the error message itself with "Uptime..." either, since it's displayed to the user already. "Failed to convert /proc/uptime to an integer" would fit the style of other blocks better.